src/main/java/seedu/address/logic/parser/prefix/PrefixHandler.java

```diff
@@ -0,0 +1,153 @@
+import seedu.address.model.tag.Tag;
+
+
+/**
+ * A utility class to handle the parsing of various prefixes and attributes for commands.
+ */
+public class PrefixHandler {
+
+    /**
+     * Identifies the type of argument based on the prefixes present in the provided {@code ArgumentMultimap}.
+     * If multiple prefixes are present, the order of checks determines which type is selected.
+     *
+     * @param argMultimap The {@code ArgumentMultimap} containing the parsed user input.
+     * @return A string representing the type of argument (e.g., "NAME", "PHONE", "EMAIL", etc.).
+     */
+    public String getArgumentType(ArgumentMultimap argMultimap) {
+        if (!argMultimap.getPreamble().isEmpty()) {
+            return "INDEX";
+        }
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            return "NAME";
+        }
+        if (argMultimap.getValue(PREFIX_ROLE).isPresent()) {
+            return "ROLE";
+        }
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            return "PHONE";
+        }
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            return "EMAIL";
+        }
+        if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+            return "ADDRESS";
+        }
+        if (!argMultimap.getAllValues(PREFIX_TAG).isEmpty()) {
+            return "TAG";
+        }
+
+        return "DEFAULT";
+    }
+
+    /**
+     * Determines the type of the given attribute object.
+     *
+     * @param attribute The attribute object.
+     * @return The type of the attribute as a string (e.g., "PHONE", "EMAIL", "ADDRESS").
+     */
+    public String getTypeOfAttribute(Object attribute) {
+        if (attribute instanceof Name) {
+            return "NAME";
+        } else if (attribute instanceof Phone) {
+            return "PHONE";
+        } else if (attribute instanceof Email) {
+            return "EMAIL";
+        } else if (attribute instanceof Address) {
+            return "ADDRESS";
+        } else if (attribute instanceof Role) {
+            return "ROLE";
+        } else if (attribute instanceof Set<?>) {
+            return "TAG";
+        } else {
+            return "ERROR";
+        }
+    }
+
+    /**
+     * Finds all persons in the given list whose attributes match the specified attribute.
+     * The matching is based on the type of attribute (e.g., phone number, email, address).
+     *
+     * @param lastShownList The list of persons to search through.
+     * @param attribute The attribute to match against (e.g., a phone number, email, or address).
+     * @return A list of persons whose attributes match the provided attribute.
+     */
+    public List<Person> findPersonByAttribute(List<Person> lastShownList, Object attribute) {
+        String type = getTypeOfAttribute(attribute);
+        List<Person> matchingPersons = new ArrayList<>();
+        for (Person person : lastShownList) {
+            switch (type) {
+            case "NAME":
```

Comment:
for the finding name feature it only finds by the first name. for list 1. Mithuna hi 2. Anna bye, [find n/hi] returns 0 persons listed but i believe from our UG it should return Mithuna hi. also [find n/ anna hi] should return both mithuna hi and anna bye rite cause we allow partial name searches?